testTruckRender.gd

```diff
@@ -0,0 +1,56 @@
+@tool
+extends Sprite2D
+
+var truckModel;
+var postProcObject;
+var lastPos: Vector2;
+var currentPos: Vector2;
+var extend_vec: Vector2;
+
+var angle_to = 0;
+
+var lastTick = 0
+
+@export var modelTint: Color = Color.RED:
+	set(val):
+		modelTint = val;
+@export_range(0, 1, 0.1) var modelTintFactor:
+	set(val):
+		modelTintFactor = val;
+		
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	truckModel = $SubViewport/testTruck
+	postProcObject = get_node("SubViewport/Camera3D/Postprocess")
+	pass # Replace with function body.
+
+func _draw():
+#	if !Engine.is_editor_hint():
+		draw_line(position, position + (extend_vec), Color.RED, 2);
+		draw_arc(position, 20, 0, deg_to_rad(-1 * angle_to), 20, Color.YELLOW, 2)
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(delta):
+	if postProcObject:
+		postProcObject.get_surface_override_material(0).set_shader_parameter("tint_color", modelTint)
+		postProcObject.get_surface_override_material(0).set_shader_parameter("tint_effect_factor", modelTintFactor)
+	if !Engine.is_editor_hint():
+		var controlNode = get_tree().get_current_scene().get_node('ControlSet')
```

Comment:
Not sure if this is negligible. But can this be done in `_ready()`, so we don't traverse the tree every frame?

Reply:
the surface override material setting is done in the process loop so that changes to the courier while in the editor are visible, else it only applies once you run the game.
The fetching of the controlNode needs to be filtered for if we're in editor mode, else it throws errors attempting to access the nonexistent ControlSet node

Reply:
I meant the `controlNode` line. Does putting the editor check in `_ready` not work? I don't know how much getting a node from the tree costs, and it could be basically nothing and not worth worrying about. But this value shouldn't change, so it should be safe to do in `_ready`, instead of in `_process`?

Reply:
Gotcha. That works, tested and changed now.